### PR TITLE
[onert] Remove copy from auto keyword

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit removes compiler and analysis warning from copying on auto keyword by adding &.

ONE-DCO-1.0-Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

### CODE REVIEW

File: runtime/onert/backend/cpu/BackendContext.cc [hunk 1]
```diff
@@ -37,7 +37,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap ret;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/backend/cpu/BackendContext.cc:40]: (Optional)
```suggestion
  for (auto &&op_ind : _data.op_order)
```

"It is safe, and in fact, preferable in generic code, to use deduction to forwarding reference, for (auto&& var : sequence)." in https://en.cppreference.com/w/cpp/language/range-for.

File: runtime/onert/backend/cpu/KernelGenerator.cc [hunk 2]
```diff
@@ -257,7 +257,7 @@ std::unique_ptr<exec::FunctionSequence> KernelGenerator::generate(ir::OperationI
   assert(_return_fn); // _return_fn must have been generated
   ret->append(std::move(_return_fn));
 
-  for (auto ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
+  for (auto &ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
```
Comment (ragmani) [inline runtime/onert/backend/cpu/KernelGenerator.cc:260]: ditto

File: runtime/onert/backend/gpu_cl/BackendContext.cc [hunk 3]
```diff
@@ -90,7 +90,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap fn_map;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/backend/gpu_cl/BackendContext.cc:93]: ditto

File: runtime/onert/backend/gpu_cl/TensorManager.cc [hunk 4]
```diff
@@ -103,10 +103,10 @@ ir::OperandIndexMap<std::shared_ptr<operand::CLTensor>> &TensorManager::nonconst
 
 void TensorManager::iterate(const std::function<void(const ir::OperandIndex &)> &fn)
 {
-  for (auto it : _nonconst_mgr->tensors())
+  for (auto &it : _nonconst_mgr->tensors())
```
Comment (ragmani) [inline runtime/onert/backend/gpu_cl/TensorManager.cc:106]: ditto

File: runtime/onert/backend/gpu_cl/TensorManager.cc [hunk 5]
```diff
@@ -103,10 +103,10 @@ ir::OperandIndexMap<std::shared_ptr<operand::CLTensor>> &TensorManager::nonconst
...
     fn(it.first);
 
-  for (auto it : _const_mgr->tensors())
+  for (auto &it : _const_mgr->tensors())
```
Comment (ragmani) [inline runtime/onert/backend/gpu_cl/TensorManager.cc:109]: ditto

File: runtime/onert/backend/ruy/BackendContext.cc [hunk 6]
```diff
@@ -37,7 +37,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap ret;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/backend/ruy/BackendContext.cc:40]: ditto

File: runtime/onert/backend/ruy/KernelGenerator.cc [hunk 7]
```diff
@@ -55,7 +55,7 @@ std::unique_ptr<exec::FunctionSequence> KernelGenerator::generate(ir::OperationI
   assert(_return_fn); // _return_fn must have been generated
   ret->append(std::move(_return_fn));
 
-  for (auto ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
+  for (auto &ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
```
Comment (ragmani) [inline runtime/onert/backend/ruy/KernelGenerator.cc:58]: ditto

File: runtime/onert/backend/trix/BackendContext.cc [hunk 8]
```diff
@@ -37,7 +37,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap ret;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/backend/trix/BackendContext.cc:40]: ditto

File: runtime/onert/backend/xnnpack/BackendContext.cc [hunk 9]
```diff
@@ -37,7 +37,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap ret;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/backend/xnnpack/BackendContext.cc:40]: ditto

File: runtime/onert/backend/xnnpack/KernelGenerator.cc [hunk 10]
```diff
@@ -69,7 +69,7 @@ std::unique_ptr<exec::FunctionSequence> KernelGenerator::generate(ir::OperationI
   assert(_return_fn); // _return_fn must have been generated
   ret->append(std::move(_return_fn));
 
-  for (auto ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
+  for (auto &ind : (op.getInputs() | ir::Remove::UNDEFINED) + op.getOutputs())
```
Comment (ragmani) [inline runtime/onert/backend/xnnpack/KernelGenerator.cc:72]: ditto

File: runtime/onert/core/include/util/Set.h [hunk 11]
```diff
@@ -145,7 +145,7 @@ template <typename Element> class Set
   Set<Element> operator-(const Set<Element> &other) const // Minus
   {
     auto ret = *this;
-    for (auto e : other)
+    for (auto &e : other)
```
Comment (ragmani) [inline runtime/onert/core/include/util/Set.h:148]: ditto

File: runtime/onert/core/src/backend/builtin/BackendContext.cc [hunk 12]
```diff
@@ -32,7 +32,7 @@ FunctionMap BackendContext::genKernels()
 {
   FunctionMap ret;
 
-  for (auto op_ind : _data.op_order)
+  for (auto &op_ind : _data.op_order)
```
Comment (ragmani) [inline runtime/onert/core/src/backend/builtin/BackendContext.cc:35]: ditto

File: runtime/onert/core/src/compiler/ExecutorFactory.cc [hunk 13]
```diff
@@ -116,7 +116,7 @@ void initializeSubgraphIOTensors(compiler::ILoweredGraph &lowered_graph,
   }
   assert(builtin_tensor_reg);
 
-  for (auto ind : indices)
+  for (auto &ind : indices)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/ExecutorFactory.cc:119]: ditto

File: runtime/onert/core/src/compiler/ExecutorFactory.cc [hunk 14]
```diff
@@ -214,7 +214,7 @@ createBackendContexts(compiler::ILoweredGraph &lgraph, bool linear_executor,
         // Add missing operands (externals)
         auto io_list = (operation.getInputs() + operation.getOutputs()) | ir::Remove::DUPLICATED |
                        ir::Remove::UNDEFINED;
-        for (auto operand_ind : io_list)
+        for (auto &operand_ind : io_list)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/ExecutorFactory.cc:217]: ditto

File: runtime/onert/core/src/compiler/ExecutorFactory.cc [hunk 15]
```diff
@@ -330,7 +330,7 @@ void ExecutorFactory::prepareMigrantTensors(compiler::ILoweredGraph &lowered_gra
     [&](const ir::OperationIndex &op_ind, const ir::IOperation &op) {
       auto lower_info = lowered_graph.lower_info().operation.getRawPtr(op_ind);
       auto &backend_ctx = backend_contexts.at(lower_info->backend());
-      for (auto ind :
+      for (auto &ind :
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/ExecutorFactory.cc:333]: ditto

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 16]
```diff
@@ -49,7 +49,7 @@ void LoweredGraph::lowerGraph(const CompilerOptions &options)
   // Build backend contexts
   auto &backend_manager = BackendManager::get();
   // Create contexts for other backends
-  for (auto backend_str : options.backend_list)
+  for (auto &backend_str : options.backend_list)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:52]: ditto

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 17]
```diff
@@ -100,9 +100,9 @@ void LoweredGraph::lowerGraph(const CompilerOptions &options)
   pass::PassRunner{}.append(std::make_unique<pass::PermutationEliminationPass>(*this)).run();
 
   VERBOSE(LoweredGraph) << "Dump after all the passes" << std::endl;
-  for (auto operand : _graph.getInputs())
+  for (auto &operand : _graph.getInputs())
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:103]: ditto

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 18]
```diff
@@ -100,9 +100,9 @@ void LoweredGraph::lowerGraph(const CompilerOptions &options)
...
     VERBOSE(LoweredGraph) << "Graph Input : " << operand << std::endl;
-  for (auto operand : _graph.getOutputs())
+  for (auto &operand : _graph.getOutputs())
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:105]: ditto

File: runtime/onert/core/src/compiler/LoweredGraph.cc [hunk 19]
```diff
@@ -135,12 +135,12 @@ void LoweredGraph::makeLowerInfo(const compiler::BackendResolver &backend_resolv
     // TODO Change setting layout of each backend at another place
     auto backend_layout = backend->config()->supportLayout(op, frontend_layout);
 
-    for (auto ind : op.getInputs() | ir::Remove::UNDEFINED)
+    for (auto &ind : op.getInputs() | ir::Remove::UNDEFINED)
```
Comment (ragmani) [inline runtime/onert/core/src/compiler/LoweredGraph.cc:138]: omitted below

**Review by ragmani — APPROVED** [review]
LGTM

**Review by glistening — APPROVED** [review]
LGTM

File: runtime/onert/backend/cpu/BackendContext.cc (hunk 1 above)
Comment (hseok-oh) [reply runtime/onert/backend/cpu/BackendContext.cc:40]: @ragmani I'll update this.

**Pull request merged**